consensus/types/src/subnet_id.rs

```diff
@@ -80,15 +80,26 @@ impl SubnetId {
         epoch: Epoch,
         spec: &ChainSpec,
     ) -> Result<(impl Iterator<Item = SubnetId>, Epoch), &'static str> {
+        // Simplify the variable name
+        let subscription_duration = spec.epochs_per_subnet_subscription;
+
         let node_id_prefix =
             (node_id >> (256 - spec.attestation_subnet_prefix_bits() as usize)).as_usize();
 
-        let subscription_event_idx = epoch.as_u64() / spec.epochs_per_subnet_subscription;
+        // NOTE: The as_u64() panics if the number is larger than u64::max_value(). This cannot be
+        // true as spec.epochs_per_subnet_subscription is a u64.
+        let node_offset = (node_id % ethereum_types::U256::from(subscription_duration)).as_u64();
+
+        // Calculate at which epoch this node needs to re-evaluate
+        let valid_until_epoch = epoch.as_u64()
+            + subscription_duration
+                .saturating_sub((epoch.as_u64() + node_offset) % subscription_duration);
+
```

Comment:
if the epoch is `0` and the offset is `1` for a subscription duration of `10` this means the offset is backwards?
Here the valid_until_epoch is `0 + 10 - (0 + 1) = 9` I would have expected this to be 1. Not claiming this to be wrong, just making sure this is the intended use